REBASE.opendatahub.md

```diff
@@ -0,0 +1,166 @@
+```
+
+### Creating a new local branch for the new rebase
+
+Branch the target release to a new branch:
+
+```
+TAG=2.2.0
+git checkout -b rebase-$TAG $TAG
+```
+
+Merge opendatahub(master) branch into the `rebase-\$TAG` branch with merge strategy ours. It discards all changes from the other branch (opendatahub/master) and create a merge commit. This leaves the content of your branch unchanged, and when you next merge with the other branch, Git will only consider changes made from this point forward. (Do not confuse this with ours conflict resolution strategy for recursive merge strategy, -X option.)
+
+```
+git merge opendatahub/master
+```
+
+This action will need to resolve some conflicts manually. Some recommentations when working in this task are:
+
+* Dockerfiles are not expected to have any merge conflicts. We should have our dsp images stored in a separate path from the kfp ones.
+* Any changes in generated files (go.mod, go.sum, package.json, package-lock.json) should always prioritize upstream changes.
+* In case of changes in backend code that diverges completelly between kfp and dsp, you should use `git blame` to find the author(s) of the changes and work together to fix the conflicts. Do not try to fix by yourself, you are not alone in this.
+
+After resolving all conflicts, remember to run a pre-flight check before going to the next task.
+
+```
+pre-commit
+make unittest
+make functest
+```
+
+### Create the Pull-Request in opendatahub-io/data-science-pipelines repository
+
+Create a PR with the result of the previous tasks with the following description: `UPSTREAM <carry>: Rebase code to kfp x.y.z`
+
+## Argo Workflows Repo
+
+If the kfp code you are rebasing uses a newer Argo workflows version, you must update opendatahub-io/argo-workflows repository.
+
+### Preparing the local repo clone
+
+Clone from a personal fork, and add the remote for upstream and opendatahub, fetching its branches:
+
+```
+git clone git@github.com:<user id>/data-science-pipelines
+git remote add --fetch argo https://github.com/argoproj/argo-workflows
+git remote add --fetch opendatahub https://github.com/opendatahub-io/argo-workflows
+```
+
+### Creating a backup branch form the dsp repo
+
+Argo Workflows git history diverges completely across versions, so it's important to create a backup branch from the current dsp repo in case we need to revert changes.
+
+```
+git checkout -b dsp-backup opendatahub/main
+```
+
+**NOTE:** Keep this branch for as long time as possible after Argo rebase, just in case we need to revert some changes.
+
+### Creating a new local branch for the new rebase
+
+Branch the target release to a new branch:
+
+```
+TAG=v3.4.17
+git checkout -b argo-upgrade $TAG
+```
+
+### Create the Pull-Request in opendatahub-io/argo-workflows repository
+
+Create a PR with the result of the previous tasks with the following description: `Upgrade argo-workflows code to x.y.z`
+
```

Comment:
```suggestion
### Create a Draft/WIP Pull Request in opendatahub-io/argo-workflows repository

**NOTE**: This is only to show the diff/changeset and accept feedback from the team before proceeding. DO NOT ACTUALLY MERGE THIS

Create a PR with the result of the previous tasks with the following description: `Upgrade argo-workflows code to x.y.z`

### Force-push Version Upgrade branch to main

Upon acceptance of the Draft PR (again, do not actually merge this), force the `opendatahub/main` branch to now match the upgrade version 'feature' branch:

```bash
git push -f origin argo-upgrade:main
```

Obviously, this will completely overwrite the git history of the `opendatahub/main` remote branch so please ensure a backup branch (`dsp-backup`) was created as instructed above


### Disclaimer / Future Work

This process this obviously very heavy-handed and destructive, and depends on there being no carries or downstream-only commits.  We should adjust the procedure to account for this

Reply:
update the codeblock escapes (\`\`\`) before accepting suggestion